2025-01-06T15:26:32.8081070Z ##[group]Run pytest tardis --tardis-regression-data=/Users/runner/work/tardis/tardis/tardis-regression-data --cov=tardis --cov-report=xml --cov-report=html --cov-append -m "not continuum"
2025-01-06T15:26:32.8082110Z [36;1mpytest tardis --tardis-regression-data=/Users/runner/work/tardis/tardis/tardis-regression-data --cov=tardis --cov-report=xml --cov-report=html --cov-append -m "not continuum"[0m
2025-01-06T15:26:32.8138910Z shell: /bin/bash -l {0}
2025-01-06T15:26:32.8139120Z env:
2025-01-06T15:26:32.8139260Z   CACHE_NUMBER: 0
2025-01-06T15:26:32.8139740Z   PYTEST_FLAGS: --tardis-regression-data=/Users/runner/work/tardis/tardis/tardis-regression-data --cov=tardis --cov-report=xml --cov-report=html --cov-append
2025-01-06T15:26:32.8140240Z   CODECOV_TOKEN: 
2025-01-06T15:26:32.8140400Z   MAMBA_ROOT_PREFIX: /Users/runner/micromamba
2025-01-06T15:26:32.8140680Z   MAMBA_EXE: /Users/runner/micromamba-bin/micromamba
2025-01-06T15:26:32.8140980Z   CONDARC: /Users/runner/work/_temp/setup-micromamba/.condarc
2025-01-06T15:26:32.8141300Z ##[endgroup]
2025-01-06T15:26:44.9937440Z ============================= test session starts ==============================
2025-01-06T15:26:44.9938130Z platform darwin -- Python 3.12.4, pytest-8.2.2, pluggy-1.5.0
2025-01-06T15:26:44.9938450Z rootdir: /Users/runner/work/tardis/tardis
2025-01-06T15:26:44.9938780Z configfile: pyproject.toml
2025-01-06T15:26:44.9939090Z plugins: cov-5.0.0, html-4.1.1, metadata-3.1.1, doctestplus-1.2.1, anyio-4.4.0
2025-01-06T15:26:44.9939510Z collected 1796 items / 2 deselected / 1794 selected
2025-01-06T15:26:44.9939730Z 
2025-01-06T15:26:47.8260340Z tardis/energy_input/tests/test_energy_source.py ..xxx                    [  0%]
2025-01-06T15:26:49.3505870Z tardis/energy_input/tests/test_gamma_ray_channel.py ...........          [  0%]
2025-01-06T15:26:50.3804200Z tardis/energy_input/tests/test_gamma_ray_grid.py xx.                     [  1%]
2025-01-06T15:26:51.4172800Z tardis/energy_input/tests/test_gamma_ray_interactions.py xx....          [  1%]
2025-01-06T15:26:51.4273450Z tardis/energy_input/tests/test_gamma_ray_packet_source.py sssssssss      [  1%]
2025-01-06T15:26:52.3151990Z tardis/energy_input/tests/test_util.py .....xxxx.....                    [  2%]
2025-01-06T15:26:52.5693360Z tardis/grid/tests/test_grid.py .                                         [  2%]
2025-01-06T15:26:52.5702340Z tardis/gui/tests/test_gui.py s                                           [  2%]
2025-01-06T15:26:55.3449490Z tardis/io/configuration/tests/test_config_reader.py ...........          [  3%]
2025-01-06T15:26:55.3553830Z tardis/io/configuration/tests/test_configuration_namespace.py .........  [  3%]
2025-01-06T15:26:55.3829600Z tardis/io/logger/tests/test_logging.py ssssssssssssssssssssssssssss      [  5%]
2025-01-06T15:27:23.1535250Z tardis/io/model/readers/tests/test_arepo_parser.py ..                    [  5%]
2025-01-06T15:27:23.1751180Z tardis/io/model/readers/tests/test_ascii_readers.py ...                  [  5%]
2025-01-06T15:27:23.1821030Z tardis/io/model/readers/tests/test_cmfgen_reader.py ..                   [  5%]
2025-01-06T15:27:23.2017560Z tardis/io/model/readers/tests/test_csvy_reader.py ....                   [  6%]
2025-01-06T15:27:23.2170340Z tardis/io/model/readers/tests/test_stella_reader.py ..                   [  6%]
2025-01-06T15:27:23.4575590Z tardis/io/tests/test_HDFWriter.py ...............                        [  7%]
2025-01-06T15:27:23.9009840Z tardis/io/tests/test_atomic.py .....                                     [  7%]
2025-01-06T15:27:23.9145070Z tardis/io/tests/test_decay.py ..                                         [  7%]
2025-01-06T15:27:24.0804350Z tardis/io/tests/test_model_reader.py ......                              [  7%]
2025-01-06T15:27:24.0851460Z tardis/model/geometry/tests/test_radial1d.py ..                          [  7%]
2025-01-06T15:27:51.1956610Z tardis/model/tests/test_base.py ........................................ [ 10%]
2025-01-06T15:27:51.9324900Z ............                                                             [ 10%]
2025-01-06T15:27:54.1890700Z tardis/model/tests/test_csvy_model.py .........                          [ 11%]
2025-01-06T15:27:54.2188110Z tardis/model/tests/test_density.py ..                                    [ 11%]
2025-01-06T15:27:55.2856990Z tardis/opacities/tests/test_opacity_solver.py .....                      [ 11%]
2025-01-06T15:27:56.3202600Z tardis/opacities/tests/test_opacity_state_numba.py .....                 [ 11%]
2025-01-06T15:27:57.9050090Z tardis/opacities/tests/test_tau_sobolev.py ..                            [ 12%]
2025-01-06T15:28:00.9530250Z tardis/plasma/equilibrium/tests/test_collisional_transitions.py ...      [ 12%]
2025-01-06T15:28:37.7874280Z tardis/plasma/equilibrium/tests/test_level_populations.py ....           [ 12%]
2025-01-06T15:28:46.3946680Z tardis/plasma/equilibrium/tests/test_rate_matrix.py .                    [ 12%]
2025-01-06T15:28:47.3693190Z tardis/plasma/tests/test_complete_plasmas.py .....................s..... [ 13%]
2025-01-06T15:28:49.3192590Z s....ssssss............s............ss.....s....ssssss............s..... [ 18%]
2025-01-06T15:28:50.7121970Z .......ss.....s....ssssss............s............ss.....s....ssssss.... [ 22%]
2025-01-06T15:28:52.9233870Z ........s............ss.....s........ss............s............ss.....s [ 26%]
2025-01-06T15:28:55.3015250Z ........ss............s............ss.....s....ssssss............s...... [ 30%]
2025-01-06T15:28:55.9486190Z ......ss.....s....ssssss............s............ss.....s.ss.ssssss..... [ 34%]
2025-01-06T15:28:58.5566460Z .......s............ss.....s....ssssss............s............ss.....s. [ 38%]
2025-01-06T15:29:01.2319980Z ...ssssss............s............ss.....s....ssssss............s....... [ 42%]
2025-01-06T15:29:04.0481300Z .....ss.....s....ssssss............s............s...........ssssss...... [ 46%]
2025-01-06T15:29:06.0505130Z ......s............s...........ssssss............s............s......... [ 50%]
2025-01-06T15:29:08.2667590Z ..ssssss............s............ss.....s....ssssss............s........ [ 54%]
2025-01-06T15:29:10.4484610Z ....ss.....s....ssssss..........................s.....s....ssss......... [ 58%]
2025-01-06T15:29:10.7413120Z ...................s.....s....ssss.......                                [ 60%]
2025-01-06T15:29:10.9567150Z tardis/plasma/tests/test_hdf_plasma.py ................................. [ 62%]
2025-01-06T15:29:10.9655860Z ......                                                                   [ 62%]
2025-01-06T15:29:11.2326940Z tardis/plasma/tests/test_nlte_excitation.py ...                          [ 62%]
2025-01-06T15:29:22.1588730Z tardis/plasma/tests/test_nlte_solver.py ..EEEx                           [ 63%]
2025-01-06T15:29:22.1693620Z tardis/plasma/tests/test_plasma_continuum.py .                           [ 63%]
2025-01-06T15:29:44.3019120Z tardis/plasma/tests/test_plasma_vboundary.py .............               [ 63%]
2025-01-06T15:29:55.7482150Z tardis/plasma/tests/test_tardis_model_density_config.py ...              [ 64%]
2025-01-06T15:29:56.3129850Z tardis/simulation/tests/test_convergence.py .....                        [ 64%]
2025-01-06T15:30:06.6680830Z tardis/simulation/tests/test_simulation.py .............                 [ 65%]
2025-01-06T15:30:06.6788190Z tardis/spectrum/tests/test_cuda_formal_integral.py sssssssssssssssssssss [ 66%]
2025-01-06T15:30:06.9660480Z ssss...ssssssssssssssssssssss                                            [ 67%]
2025-01-06T15:30:06.9700760Z tardis/spectrum/tests/test_luminosity.py ....                            [ 68%]
2025-01-06T15:30:07.8568380Z tardis/spectrum/tests/test_numba_formal_integral.py .................... [ 69%]
2025-01-06T15:30:07.8687360Z ..........                                                               [ 69%]
2025-01-06T15:30:08.0634250Z tardis/spectrum/tests/test_spectrum.py ................................. [ 71%]
2025-01-06T15:30:08.0989430Z .........                                                                [ 72%]
2025-01-06T15:30:27.9879090Z tardis/spectrum/tests/test_spectrum_solver.py ....                       [ 72%]
2025-01-06T15:31:26.3737500Z tardis/tests/test_tardis_full.py ....                                    [ 72%]
2025-01-06T15:32:55.5890420Z tardis/tests/test_tardis_full_formal_integral.py ................        [ 73%]
2025-01-06T15:32:55.6575920Z tardis/tests/test_util.py .............................................  [ 75%]
2025-01-06T15:32:58.9524990Z tardis/transport/montecarlo/tests/test_base.py x.....................    [ 77%]
2025-01-06T15:33:08.0044940Z tardis/transport/montecarlo/tests/test_interaction.py .......            [ 77%]
2025-01-06T15:33:08.3230320Z tardis/transport/montecarlo/tests/test_macro_atom.py ....                [ 77%]
2025-01-06T15:33:08.3617030Z tardis/transport/montecarlo/tests/test_montecarlo.py sssssssssssssssssss [ 78%]
2025-01-06T15:33:08.4247920Z sssssssssssssssssssssssssssssssssssssssssssss                            [ 81%]
2025-01-06T15:33:20.9523160Z tardis/transport/montecarlo/tests/test_montecarlo_main_loop.py x..       [ 81%]
2025-01-06T15:33:20.9721690Z tardis/transport/montecarlo/tests/test_nonhomologous.py ...              [ 81%]
2025-01-06T15:33:24.7644200Z tardis/transport/montecarlo/tests/test_numba_interface.py .....x.        [ 82%]
2025-01-06T15:33:25.3783240Z tardis/transport/montecarlo/tests/test_opacities.py ...............      [ 82%]
2025-01-06T15:33:26.4959480Z tardis/transport/montecarlo/tests/test_packet.py .................xXXXXX [ 84%]
2025-01-06T15:33:26.6279990Z XX.........                                                              [ 84%]
2025-01-06T15:33:26.6751830Z tardis/transport/montecarlo/tests/test_packet_source.py ........         [ 85%]
2025-01-06T15:33:26.9071310Z tardis/transport/montecarlo/tests/test_rpacket_last_interaction_tracker.py . [ 85%]
2025-01-06T15:33:27.3027990Z .....                                                                    [ 85%]
2025-01-06T15:33:38.7545400Z tardis/transport/montecarlo/tests/test_rpacket_tracker.py ......         [ 85%]
2025-01-06T15:33:38.7769940Z tardis/transport/montecarlo/tests/test_single_packet_loop.py xxx         [ 86%]
2025-01-06T15:33:38.7805920Z tardis/transport/montecarlo/tests/test_tracker_utils.py ..               [ 86%]
2025-01-06T15:33:40.1226000Z tardis/transport/montecarlo/tests/test_vpacket.py ..x.                   [ 86%]
2025-01-06T15:33:40.1390400Z tardis/transport/montecarlo/tests/test_weighted_packet_source.py ....    [ 86%]
2025-01-06T15:33:43.0556800Z tardis/transport/montecarlo/tests/test_weighted_packet_source_integration.py . [ 86%]
2025-01-06T15:33:43.0558710Z                                                                          [ 86%]
2025-01-06T15:33:43.6662130Z tardis/transport/tests/test_doppler_factor.py ..................         [ 87%]
2025-01-06T15:33:46.1787220Z tardis/visualization/tools/tests/test_convergence_plot.py .............. [ 88%]
2025-01-06T15:33:47.8677860Z ......                                                                   [ 88%]
2025-01-06T15:33:51.5969100Z tardis/visualization/tools/tests/test_liv_plot.py ...................... [ 90%]
2025-01-06T15:33:55.9088820Z .............................                                            [ 91%]
2025-01-06T15:33:58.8084270Z tardis/visualization/tools/tests/test_rpacket_plot.py .................. [ 92%]
2025-01-06T15:34:06.1367650Z ....                                                                     [ 92%]
2025-01-06T15:34:10.6886720Z tardis/visualization/tools/tests/test_sdec_plot.py ..................... [ 94%]
2025-01-06T15:34:17.2391120Z ..............................                                           [ 95%]
2025-01-06T15:34:17.5838260Z tardis/visualization/widgets/tests/test_custom_abundance.py ........ss.. [ 96%]
2025-01-06T15:34:17.6227010Z ....                                                                     [ 96%]
2025-01-06T15:34:18.3785650Z tardis/visualization/widgets/tests/test_line_info.py ................... [ 97%]
2025-01-06T15:34:18.8427230Z ..................                                                       [ 98%]
2025-01-06T15:37:45.9632670Z tardis/visualization/widgets/tests/test_shell_info.py .................. [ 99%]
2025-01-06T15:37:54.5499940Z ......                                                                   [100%]
2025-01-06T15:37:54.5503150Z 
2025-01-06T15:37:54.5503540Z ==================================== ERRORS ====================================
2025-01-06T15:37:54.5504700Z _________ ERROR at setup of test_critical_case_dilution_factor_1_root __________
2025-01-06T15:37:54.5505520Z 
2025-01-06T15:37:54.5509580Z tardis_model_config_nlte_root = {'tardis_config_version': 'v1.0', 'supernova': {'luminosity_requested': <Quantity 4.31528298e+42 erg / s>, 'time_explo....0, 'enable_biasing': False, 'virtual_packet_logging': False}}, 'config_dirname': 'tardis/io/configuration/tests/data'}
2025-01-06T15:37:54.5513650Z nlte_raw_model_root = <tardis.model.base.SimulationState object at 0x137cdf440>
2025-01-06T15:37:54.5517800Z nlte_atom_data = <Atomic Data UUID=None MD5=None Lines=69056 Levels=26012>
2025-01-06T15:37:54.5522580Z 
2025-01-06T15:37:54.5522880Z     @pytest.fixture
2025-01-06T15:37:54.5523510Z     def nlte_raw_plasma_dilution_factor_1_root(
2025-01-06T15:37:54.5524010Z         tardis_model_config_nlte_root, nlte_raw_model_root, nlte_atom_data
2025-01-06T15:37:54.5524490Z     ):
2025-01-06T15:37:54.5525300Z         """
2025-01-06T15:37:54.5594320Z         Plasma assembled with dilution factors set to 1.0.
2025-01-06T15:37:54.5595010Z         """
2025-01-06T15:37:54.5595540Z         new_dilution_factor = np.ones_like(nlte_raw_model_root.dilution_factor)
2025-01-06T15:37:54.5596110Z         nlte_raw_model_root.dilution_factor = new_dilution_factor
2025-01-06T15:37:54.5596580Z >       plasma = assemble_plasma(
2025-01-06T15:37:54.5596950Z             tardis_model_config_nlte_root, nlte_raw_model_root, nlte_atom_data
2025-01-06T15:37:54.5597400Z         )
2025-01-06T15:37:54.5608500Z 
2025-01-06T15:37:54.5640730Z tardis/plasma/tests/test_nlte_solver.py:232: 
2025-01-06T15:37:54.5641580Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:37:54.5642160Z tardis/plasma/assembly/legacy_assembly.py:37: in assemble_plasma
2025-01-06T15:37:54.5642620Z     return plasma_solver_factory.assemble(
2025-01-06T15:37:54.5643120Z tardis/plasma/assembly/base.py:614: in assemble
2025-01-06T15:37:54.5643480Z     return BasePlasma(
2025-01-06T15:37:54.5643870Z tardis/io/util.py:195: in __new__
2025-01-06T15:37:54.5644270Z     instance.__init__(*args, **kwargs)
2025-01-06T15:37:54.5644570Z tardis/plasma/base.py:40: in __init__
2025-01-06T15:37:54.5644980Z     self.update(**kwargs)
2025-01-06T15:37:54.5645220Z tardis/plasma/base.py:192: in update
2025-01-06T15:37:54.5647850Z     self.plasma_properties_dict[module_name].update()
2025-01-06T15:37:54.5648230Z tardis/plasma/properties/base.py:110: in update
2025-01-06T15:37:54.5648570Z     self.calculate(*self._get_input_values()),
2025-01-06T15:37:54.5649020Z tardis/plasma/properties/continuum_processes/rates.py:231: in calculate
2025-01-06T15:37:54.5649550Z     alpha_sp_E = cumulative_integrate_array_by_blocks(
2025-01-06T15:37:54.5650120Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:468: in _compile_for_args
2025-01-06T15:37:54.5651110Z     error_rewrite(e, 'typing')
2025-01-06T15:37:54.5651400Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:37:54.5651610Z 
2025-01-06T15:37:54.5654470Z e = TypingError('Failed in nopython mode pipeline (step: nopython frontend)\n\x1b[1m\x1b[1m\x1b[1m\x1b[1mFailed in nopytho...[1m            integrated[start:stop, col] = numba_cumulative_trapezoid(\n\x1b[0m            \x1b[1m^\x1b[0m\x1b[0m\n')
2025-01-06T15:37:54.5655190Z issue_type = 'typing'
2025-01-06T15:37:54.5655340Z 
2025-01-06T15:37:54.5655460Z     def error_rewrite(e, issue_type):
2025-01-06T15:37:54.5655690Z         """
2025-01-06T15:37:54.5655960Z         Rewrite and raise Exception `e` with help supplied based on the
2025-01-06T15:37:54.5656260Z         specified issue_type.
2025-01-06T15:37:54.5666730Z         """
2025-01-06T15:37:54.5666980Z         if config.SHOW_HELP:
2025-01-06T15:37:54.5667320Z             help_msg = errors.error_extras[issue_type]
2025-01-06T15:37:54.5667670Z             e.patch_message('\n'.join((str(e).rstrip(), help_msg)))
2025-01-06T15:37:54.5668050Z         if config.FULL_TRACEBACKS:
2025-01-06T15:37:54.5668280Z             raise e
2025-01-06T15:37:54.5668520Z         else:
2025-01-06T15:37:54.5668670Z >           raise e.with_traceback(None)
2025-01-06T15:37:54.5669130Z E           numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5705790Z E           [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5707630Z E           [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:37:54.5708880Z E            
2025-01-06T15:37:54.5709620Z E            >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:37:54.5710470Z E            
2025-01-06T15:37:54.5711070Z E           There are 2 candidate implementations:
2025-01-06T15:37:54.5711960Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.5712950Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.5714380Z E               With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.5715580Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5716640Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5718000Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.5719880Z E               
2025-01-06T15:37:54.5720650Z E               >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.5721620Z E               
2025-01-06T15:37:54.5722160Z E              There are 2 candidate implementations:
2025-01-06T15:37:54.5723110Z E              [1m      - Of which 2 did not match due to:
2025-01-06T15:37:54.5724100Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5725670Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.5726830Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5728570Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5732340Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5733420Z E              [0m
2025-01-06T15:37:54.5734410Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.5736130Z E              [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.5737470Z E              [0m
2025-01-06T15:37:54.5737920Z E              [1m
2025-01-06T15:37:54.5739170Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.5740450Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.5741450Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.5742390Z E              [0m            [1m^[0m[0m
2025-01-06T15:37:54.5743080Z E           [0m
2025-01-06T15:37:54.5744110Z E             raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.5745540Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.5747820Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.5749460Z E               With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.5750040Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5750500Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5751040Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.5751360Z E               
2025-01-06T15:37:54.5751610Z E               >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.5751900Z E               
2025-01-06T15:37:54.5752140Z E              There are 2 candidate implementations:
2025-01-06T15:37:54.5752430Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.5752780Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5753240Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.5753630Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5754220Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5754850Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5755310Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.5755630Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5756680Z E                      With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:37:54.5757190Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5757710Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5758320Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5758750Z E              [0m
2025-01-06T15:37:54.5759110Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.5759760Z E              [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.5760280Z E              [0m
2025-01-06T15:37:54.5760490Z E              [1m
2025-01-06T15:37:54.5760940Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.5761430Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.5761860Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.5762160Z E              [0m            [1m^[0m[0m
2025-01-06T15:37:54.5762760Z E           [0m
2025-01-06T15:37:54.5763200Z E             raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.5763830Z E           [0m
2025-01-06T15:37:54.5764210Z E           [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:37:54.5764920Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:37:54.5765530Z E           [0m
2025-01-06T15:37:54.5765740Z E           [1m
2025-01-06T15:37:54.5766160Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:37:54.5766970Z E           [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:37:54.5767290Z E               <source elided>
2025-01-06T15:37:54.5767500Z E           
2025-01-06T15:37:54.5767900Z E           [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:37:54.5768220Z E           [0m    [1m^[0m[0m
2025-01-06T15:37:54.5768470Z E           
2025-01-06T15:37:54.5768910Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.5769750Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.5770210Z E           [0m
2025-01-06T15:37:54.5770700Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.5771480Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.5771990Z E           [0m
2025-01-06T15:37:54.5772150Z E           [1m
2025-01-06T15:37:54.5772480Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:37:54.5772960Z E           [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:37:54.5773240Z E               <source elided>
2025-01-06T15:37:54.5773460Z E           
2025-01-06T15:37:54.5773730Z E           [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:37:54.5774640Z E           [0m            [1m^[0m[0m
2025-01-06T15:37:54.5775050Z 
2025-01-06T15:37:54.5775700Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:409: TypingError
2025-01-06T15:37:54.5777390Z ---------------------------- Captured stdout setup -----------------------------
2025-01-06T15:37:54.5778470Z [[1mtardis.io.model.parse_density_configuration[0m][[1;33mWARNING[0m]  
2025-01-06T15:37:54.5780010Z 	Number of density points larger than number of shells. Assuming inner point irrelevant ([1mparse_density_configuration.py[0m:114)
2025-01-06T15:37:54.5781320Z ------------------------------ Captured log setup ------------------------------
2025-01-06T15:37:54.5782940Z WARNING  tardis.io.model.parse_density_configuration:parse_density_configuration.py:114 Number of density points larger than number of shells. Assuming inner point irrelevant
2025-01-06T15:37:54.5784560Z __________ ERROR at setup of test_critical_case_dilution_factor_1_lu ___________
2025-01-06T15:37:54.5785160Z 
2025-01-06T15:37:54.5786740Z tardis_model_config_nlte_lu = {'tardis_config_version': 'v1.0', 'supernova': {'luminosity_requested': <Quantity 4.31528298e+42 erg / s>, 'time_explo....0, 'enable_biasing': False, 'virtual_packet_logging': False}}, 'config_dirname': 'tardis/io/configuration/tests/data'}
2025-01-06T15:37:54.5789780Z nlte_raw_model_lu = <tardis.model.base.SimulationState object at 0x169aeefc0>
2025-01-06T15:37:54.5790780Z nlte_atom_data = <Atomic Data UUID=None MD5=None Lines=69056 Levels=26012>
2025-01-06T15:37:54.5791360Z 
2025-01-06T15:37:54.5791530Z     @pytest.fixture
2025-01-06T15:37:54.5791990Z     def nlte_raw_plasma_dilution_factor_1_lu(
2025-01-06T15:37:54.5792730Z         tardis_model_config_nlte_lu, nlte_raw_model_lu, nlte_atom_data
2025-01-06T15:37:54.5793580Z     ):
2025-01-06T15:37:54.5794160Z         """
2025-01-06T15:37:54.5794890Z         Plasma assembled with dilution factors set to 1.0.
2025-01-06T15:37:54.5795780Z         """
2025-01-06T15:37:54.5796750Z         new_dilution_factor = np.ones_like(nlte_raw_model_lu.dilution_factor)
2025-01-06T15:37:54.5798170Z         nlte_raw_model_lu.dilution_factor = new_dilution_factor
2025-01-06T15:37:54.5799210Z >       plasma = assemble_plasma(
2025-01-06T15:37:54.5800150Z             tardis_model_config_nlte_lu, nlte_raw_model_lu, nlte_atom_data
2025-01-06T15:37:54.5802140Z         )
2025-01-06T15:37:54.5802340Z 
2025-01-06T15:37:54.5802670Z tardis/plasma/tests/test_nlte_solver.py:247: 
2025-01-06T15:37:54.5803460Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:37:54.5804390Z tardis/plasma/assembly/legacy_assembly.py:37: in assemble_plasma
2025-01-06T15:37:54.5805280Z     return plasma_solver_factory.assemble(
2025-01-06T15:37:54.5806540Z tardis/plasma/assembly/base.py:614: in assemble
2025-01-06T15:37:54.5807440Z     return BasePlasma(
2025-01-06T15:37:54.5807690Z tardis/io/util.py:195: in __new__
2025-01-06T15:37:54.5807990Z     instance.__init__(*args, **kwargs)
2025-01-06T15:37:54.5808560Z tardis/plasma/base.py:40: in __init__
2025-01-06T15:37:54.5809500Z     self.update(**kwargs)
2025-01-06T15:37:54.5810030Z tardis/plasma/base.py:192: in update
2025-01-06T15:37:54.5810700Z     self.plasma_properties_dict[module_name].update()
2025-01-06T15:37:54.5811480Z tardis/plasma/properties/base.py:110: in update
2025-01-06T15:37:54.5813000Z     self.calculate(*self._get_input_values()),
2025-01-06T15:37:54.5813980Z tardis/plasma/properties/continuum_processes/rates.py:231: in calculate
2025-01-06T15:37:54.5814450Z     alpha_sp_E = cumulative_integrate_array_by_blocks(
2025-01-06T15:37:54.5814980Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:468: in _compile_for_args
2025-01-06T15:37:54.5815440Z     error_rewrite(e, 'typing')
2025-01-06T15:37:54.5815710Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:37:54.5815930Z 
2025-01-06T15:37:54.5816470Z e = TypingError('Failed in nopython mode pipeline (step: nopython frontend)\n\x1b[1m\x1b[1m\x1b[1m\x1b[1mFailed in nopytho...[1m            integrated[start:stop, col] = numba_cumulative_trapezoid(\n\x1b[0m            \x1b[1m^\x1b[0m\x1b[0m\n')
2025-01-06T15:37:54.5818320Z issue_type = 'typing'
2025-01-06T15:37:54.5818460Z 
2025-01-06T15:37:54.5818570Z     def error_rewrite(e, issue_type):
2025-01-06T15:37:54.5818750Z         """
2025-01-06T15:37:54.5819010Z         Rewrite and raise Exception `e` with help supplied based on the
2025-01-06T15:37:54.5819370Z         specified issue_type.
2025-01-06T15:37:54.5819960Z         """
2025-01-06T15:37:54.5820470Z         if config.SHOW_HELP:
2025-01-06T15:37:54.5820720Z             help_msg = errors.error_extras[issue_type]
2025-01-06T15:37:54.5821110Z             e.patch_message('\n'.join((str(e).rstrip(), help_msg)))
2025-01-06T15:37:54.5821690Z         if config.FULL_TRACEBACKS:
2025-01-06T15:37:54.5822500Z             raise e
2025-01-06T15:37:54.5822810Z         else:
2025-01-06T15:37:54.5824110Z >           raise e.with_traceback(None)
2025-01-06T15:37:54.5825590Z E           numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5826730Z E           [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5828580Z E           [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:37:54.5829060Z E            
2025-01-06T15:37:54.5830220Z E            >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:37:54.5830880Z E            
2025-01-06T15:37:54.5831520Z E           There are 2 candidate implementations:
2025-01-06T15:37:54.5832280Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.5833040Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.5833920Z E               With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.5834900Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5835320Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5836430Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.5837690Z E               
2025-01-06T15:37:54.5838470Z E               >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.5841150Z E               
2025-01-06T15:37:54.5841620Z E              There are 2 candidate implementations:
2025-01-06T15:37:54.5842490Z E              [1m      - Of which 2 did not match due to:
2025-01-06T15:37:54.5842940Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5843470Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.5844530Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5845500Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5848930Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5849570Z E              [0m
2025-01-06T15:37:54.5850540Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.5853490Z E              [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.5855310Z E              [0m
2025-01-06T15:37:54.5856090Z E              [1m
2025-01-06T15:37:54.5857680Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.5859110Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.5859690Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.5860760Z E              [0m            [1m^[0m[0m
2025-01-06T15:37:54.5862270Z E           [0m
2025-01-06T15:37:54.5863040Z E             raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.5864150Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.5867140Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.5867780Z E               With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.5870020Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5870710Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5871590Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.5872150Z E               
2025-01-06T15:37:54.5873160Z E               >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.5874860Z E               
2025-01-06T15:37:54.5875610Z E              There are 2 candidate implementations:
2025-01-06T15:37:54.5877070Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.5878480Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5879430Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.5881730Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5882610Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5883410Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5883980Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.5885980Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5914620Z E                      With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:37:54.5915420Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5916100Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5917970Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5920570Z E              [0m
2025-01-06T15:37:54.5921570Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.5925220Z E              [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.5926690Z E              [0m
2025-01-06T15:37:54.5926910Z E              [1m
2025-01-06T15:37:54.5927330Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.5927870Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.5928190Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.5930210Z E              [0m            [1m^[0m[0m
2025-01-06T15:37:54.5930800Z E           [0m
2025-01-06T15:37:54.5931210Z E             raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.5932080Z E           [0m
2025-01-06T15:37:54.5932380Z E           [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:37:54.5936820Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:37:54.5938630Z E           [0m
2025-01-06T15:37:54.5938850Z E           [1m
2025-01-06T15:37:54.5939220Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:37:54.5939690Z E           [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:37:54.5939940Z E               <source elided>
2025-01-06T15:37:54.5940120Z E           
2025-01-06T15:37:54.5940410Z E           [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:37:54.5942360Z E           [0m    [1m^[0m[0m
2025-01-06T15:37:54.5943060Z E           
2025-01-06T15:37:54.5943510Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.5944260Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.5945910Z E           [0m
2025-01-06T15:37:54.5947750Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.5951150Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.5951690Z E           [0m
2025-01-06T15:37:54.5952560Z E           [1m
2025-01-06T15:37:54.5953910Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:37:54.5955570Z E           [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:37:54.5956360Z E               <source elided>
2025-01-06T15:37:54.5956570Z E           
2025-01-06T15:37:54.5956870Z E           [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:37:54.5957220Z E           [0m            [1m^[0m[0m
2025-01-06T15:37:54.5959440Z 
2025-01-06T15:37:54.5959680Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:409: TypingError
2025-01-06T15:37:54.5960100Z ---------------------------- Captured stdout setup -----------------------------
2025-01-06T15:37:54.5960510Z [[1mtardis.io.model.parse_density_configuration[0m][[1;33mWARNING[0m]  
2025-01-06T15:37:54.5961070Z 	Number of density points larger than number of shells. Assuming inner point irrelevant ([1mparse_density_configuration.py[0m:114)
2025-01-06T15:37:54.5961570Z [[1mtardis.plasma.assembly.base[0m][[1;33mWARNING[0m]  
2025-01-06T15:37:54.5961980Z 	LU solver will be inaccurate for NLTE excitation, proceed with caution. ([1mbase.py[0m:489)
2025-01-06T15:37:54.5962360Z ------------------------------ Captured log setup ------------------------------
2025-01-06T15:37:54.5962950Z WARNING  tardis.io.model.parse_density_configuration:parse_density_configuration.py:114 Number of density points larger than number of shells. Assuming inner point irrelevant
2025-01-06T15:37:54.5963660Z WARNING  tardis.plasma.assembly.base:base.py:489 LU solver will be inaccurate for NLTE excitation, proceed with caution.
2025-01-06T15:37:54.5964130Z _________ ERROR at setup of test_critical_case_dilution_factor_0_root __________
2025-01-06T15:37:54.5964340Z 
2025-01-06T15:37:54.5964890Z tardis_model_config_nlte_root = {'tardis_config_version': 'v1.0', 'supernova': {'luminosity_requested': <Quantity 4.31528298e+42 erg / s>, 'time_explo....0, 'enable_biasing': False, 'virtual_packet_logging': False}}, 'config_dirname': 'tardis/io/configuration/tests/data'}
2025-01-06T15:37:54.5965690Z nlte_raw_model_root = <tardis.model.base.SimulationState object at 0x31aa905f0>
2025-01-06T15:37:54.5966040Z nlte_atom_data = <Atomic Data UUID=None MD5=None Lines=69056 Levels=26012>
2025-01-06T15:37:54.5966250Z 
2025-01-06T15:37:54.5966430Z     @pytest.fixture
2025-01-06T15:37:54.5966610Z     def nlte_raw_plasma_dilution_factor_0_root(
2025-01-06T15:37:54.5966890Z         tardis_model_config_nlte_root, nlte_raw_model_root, nlte_atom_data
2025-01-06T15:37:54.5967140Z     ):
2025-01-06T15:37:54.5967260Z         """
2025-01-06T15:37:54.5967430Z         Plasma assembled with dilution factors set to 0.0.
2025-01-06T15:37:54.5967640Z         """
2025-01-06T15:37:54.5967850Z         new_dilution_factor = np.zeros_like(nlte_raw_model_root.dilution_factor)
2025-01-06T15:37:54.5968170Z         nlte_raw_model_root.dilution_factor = new_dilution_factor
2025-01-06T15:37:54.5968420Z >       plasma = assemble_plasma(
2025-01-06T15:37:54.5968680Z             tardis_model_config_nlte_root, nlte_raw_model_root, nlte_atom_data
2025-01-06T15:37:54.5968930Z         )
2025-01-06T15:37:54.5969000Z 
2025-01-06T15:37:54.5969080Z tardis/plasma/tests/test_nlte_solver.py:262: 
2025-01-06T15:37:54.5969340Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:37:54.5969660Z tardis/plasma/assembly/legacy_assembly.py:37: in assemble_plasma
2025-01-06T15:37:54.5969950Z     return plasma_solver_factory.assemble(
2025-01-06T15:37:54.5970200Z tardis/plasma/assembly/base.py:614: in assemble
2025-01-06T15:37:54.5970420Z     return BasePlasma(
2025-01-06T15:37:54.5970590Z tardis/io/util.py:195: in __new__
2025-01-06T15:37:54.5970780Z     instance.__init__(*args, **kwargs)
2025-01-06T15:37:54.5970980Z tardis/plasma/base.py:40: in __init__
2025-01-06T15:37:54.5971170Z     self.update(**kwargs)
2025-01-06T15:37:54.5971340Z tardis/plasma/base.py:192: in update
2025-01-06T15:37:54.5971580Z     self.plasma_properties_dict[module_name].update()
2025-01-06T15:37:54.5971840Z tardis/plasma/properties/base.py:110: in update
2025-01-06T15:37:54.5972080Z     self.calculate(*self._get_input_values()),
2025-01-06T15:37:54.5972380Z tardis/plasma/properties/continuum_processes/rates.py:231: in calculate
2025-01-06T15:37:54.5972720Z     alpha_sp_E = cumulative_integrate_array_by_blocks(
2025-01-06T15:37:54.5973140Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:468: in _compile_for_args
2025-01-06T15:37:54.5973710Z     error_rewrite(e, 'typing')
2025-01-06T15:37:54.5973920Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:37:54.5974080Z 
2025-01-06T15:37:54.5974560Z e = TypingError('Failed in nopython mode pipeline (step: nopython frontend)\n\x1b[1m\x1b[1m\x1b[1m\x1b[1mFailed in nopytho...[1m            integrated[start:stop, col] = numba_cumulative_trapezoid(\n\x1b[0m            \x1b[1m^\x1b[0m\x1b[0m\n')
2025-01-06T15:37:54.5975160Z issue_type = 'typing'
2025-01-06T15:37:54.5975250Z 
2025-01-06T15:37:54.5975320Z     def error_rewrite(e, issue_type):
2025-01-06T15:37:54.5975500Z         """
2025-01-06T15:37:54.5975690Z         Rewrite and raise Exception `e` with help supplied based on the
2025-01-06T15:37:54.5975960Z         specified issue_type.
2025-01-06T15:37:54.5976120Z         """
2025-01-06T15:37:54.5976250Z         if config.SHOW_HELP:
2025-01-06T15:37:54.5976440Z             help_msg = errors.error_extras[issue_type]
2025-01-06T15:37:54.5976700Z             e.patch_message('\n'.join((str(e).rstrip(), help_msg)))
2025-01-06T15:37:54.5976940Z         if config.FULL_TRACEBACKS:
2025-01-06T15:37:54.5977110Z             raise e
2025-01-06T15:37:54.5977240Z         else:
2025-01-06T15:37:54.5977390Z >           raise e.with_traceback(None)
2025-01-06T15:37:54.5977710Z E           numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5978190Z E           [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5978660Z E           [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:37:54.5978980Z E            
2025-01-06T15:37:54.5979190Z E            >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:37:54.5979520Z E            
2025-01-06T15:37:54.5979680Z E           There are 2 candidate implementations:
2025-01-06T15:37:54.5979960Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.5980280Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.5980710Z E               With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.5981090Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5981430Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5982500Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.5982850Z E               
2025-01-06T15:37:54.5983100Z E               >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.5983390Z E               
2025-01-06T15:37:54.5983560Z E              There are 2 candidate implementations:
2025-01-06T15:37:54.5983830Z E              [1m      - Of which 2 did not match due to:
2025-01-06T15:37:54.5984150Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5984600Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.5984990Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5985520Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5986130Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5986530Z E              [0m
2025-01-06T15:37:54.5986830Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.5987430Z E              [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.5988440Z E              [0m
2025-01-06T15:37:54.5988590Z E              [1m
2025-01-06T15:37:54.5988950Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.5989340Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.5989620Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.5989870Z E              [0m            [1m^[0m[0m
2025-01-06T15:37:54.5990070Z E           [0m
2025-01-06T15:37:54.5990410Z E             raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.5990840Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.5991160Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.5991580Z E               With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.5991950Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5992290Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.5992770Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.5993090Z E               
2025-01-06T15:37:54.5993340Z E               >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.5993620Z E               
2025-01-06T15:37:54.5993790Z E              There are 2 candidate implementations:
2025-01-06T15:37:54.5994060Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.5994550Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5995010Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.5995400Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5995910Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5996510Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5996970Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.5997290Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.5997740Z E                      With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:37:54.5998150Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.5998660Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.5999250Z E                raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.5999630Z E              [0m
2025-01-06T15:37:54.5999930Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.6000500Z E              [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.6000910Z E              [0m
2025-01-06T15:37:54.6001070Z E              [1m
2025-01-06T15:37:54.6001420Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.6001820Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.6002990Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.6003240Z E              [0m            [1m^[0m[0m
2025-01-06T15:37:54.6003440Z E           [0m
2025-01-06T15:37:54.6003780Z E             raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.6004160Z E           [0m
2025-01-06T15:37:54.6004450Z E           [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:37:54.6005060Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:37:54.6005500Z E           [0m
2025-01-06T15:37:54.6005640Z E           [1m
2025-01-06T15:37:54.6005960Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:37:54.6006340Z E           [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:37:54.6006550Z E               <source elided>
2025-01-06T15:37:54.6006720Z E           
2025-01-06T15:37:54.6006960Z E           [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:37:54.6007240Z E           [0m    [1m^[0m[0m
2025-01-06T15:37:54.6007410Z E           
2025-01-06T15:37:54.6007780Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.6008470Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.6008920Z E           [0m
2025-01-06T15:37:54.6009310Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.6009990Z E           [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.6010760Z E           [0m
2025-01-06T15:37:54.6010920Z E           [1m
2025-01-06T15:37:54.6011240Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:37:54.6011670Z E           [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:37:54.6011950Z E               <source elided>
2025-01-06T15:37:54.6012110Z E           
2025-01-06T15:37:54.6012350Z E           [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:37:54.6012630Z E           [0m            [1m^[0m[0m
2025-01-06T15:37:54.6012750Z 
2025-01-06T15:37:54.6013460Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:409: TypingError
2025-01-06T15:37:54.6013950Z ---------------------------- Captured stdout setup -----------------------------
2025-01-06T15:37:54.6014380Z [[1mtardis.io.model.parse_density_configuration[0m][[1;33mWARNING[0m]  
2025-01-06T15:37:54.6015690Z 	Number of density points larger than number of shells. Assuming inner point irrelevant ([1mparse_density_configuration.py[0m:114)
2025-01-06T15:37:54.6016180Z ------------------------------ Captured log setup ------------------------------
2025-01-06T15:37:54.6016760Z WARNING  tardis.io.model.parse_density_configuration:parse_density_configuration.py:114 Number of density points larger than number of shells. Assuming inner point irrelevant
2025-01-06T15:37:54.6017320Z =============================== warnings summary ===============================
2025-01-06T15:37:54.6017570Z tardis/io/configuration/config_validator.py:6
2025-01-06T15:37:54.6017800Z tardis/io/configuration/config_validator.py:6
2025-01-06T15:37:54.6018190Z   /Users/runner/work/tardis/tardis/tardis/io/configuration/config_validator.py:6: DeprecationWarning:
2025-01-06T15:37:54.6018560Z   
2025-01-06T15:37:54.6019660Z   jsonschema.RefResolver is deprecated as of v4.18.0, in favor of the https://github.com/python-jsonschema/referencing library, which provides more compliant referencing behavior as well as more flexible APIs for customization. A future release will remove RefResolver. Please file a feature request (on referencing) if you are missing an API for the kind of customization you need.
2025-01-06T15:37:54.6021020Z 
2025-01-06T15:37:54.6021210Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/jupyter_client/connect.py:22
2025-01-06T15:37:54.6021740Z   /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/jupyter_client/connect.py:22: DeprecationWarning:
2025-01-06T15:37:54.6022140Z   
2025-01-06T15:37:54.6022330Z   Jupyter is migrating its paths to use standard platformdirs
2025-01-06T15:37:54.6022670Z   given by the platformdirs library.  To remove this warning and
2025-01-06T15:37:54.6023010Z   see the appropriate new directories, set the environment variable
2025-01-06T15:37:54.6023320Z   `JUPYTER_PLATFORM_DIRS=1` and then run `jupyter --paths`.
2025-01-06T15:37:54.6023640Z   The use of platformdirs will be the default in `jupyter_core` v6
2025-01-06T15:37:54.6023830Z 
2025-01-06T15:37:54.6051940Z tardis/io/tests/test_HDFWriter.py::test_multi_index_write
2025-01-06T15:37:54.6052470Z tardis/visualization/widgets/tests/test_shell_info.py::TestHDFShellInfo::test_shells_data
2025-01-06T15:37:54.6052920Z   /Users/runner/work/tardis/tardis/tardis/io/util.py:276: PerformanceWarning:
2025-01-06T15:37:54.6053210Z   
2025-01-06T15:37:54.6053320Z   
2025-01-06T15:37:54.6053550Z   your performance may suffer as PyTables will pickle object types that it cannot
2025-01-06T15:37:54.6053990Z   map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index([0], dtype='int64')]
2025-01-06T15:37:54.6054270Z 
2025-01-06T15:37:54.6054440Z tardis/io/tests/test_model_reader.py::test_read_cmfgen_isotope_abundances
2025-01-06T15:37:54.6054830Z tardis/plasma/tests/test_tardis_model_density_config.py::test_electron_densities
2025-01-06T15:37:54.6056290Z tardis/plasma/tests/test_tardis_model_density_config.py::test_isotope_number_densities
2025-01-06T15:37:54.6056700Z tardis/plasma/tests/test_tardis_model_density_config.py::test_t_rad
2025-01-06T15:37:54.6057160Z   /Users/runner/work/tardis/tardis/tardis/io/model/readers/cmfgen_deprecated.py:82: DeprecationWarning:
2025-01-06T15:37:54.6057530Z   
2025-01-06T15:37:54.6057680Z   The current CMFGEN model parser is deprecated
2025-01-06T15:37:54.6057830Z 
2025-01-06T15:37:54.6057910Z tardis/io/tests/test_model_reader.py: 1 warning
2025-01-06T15:37:54.6058210Z tardis/plasma/tests/test_tardis_model_density_config.py: 9 warnings
2025-01-06T15:37:54.6058660Z   /Users/runner/work/tardis/tardis/tardis/io/model/readers/cmfgen_deprecated.py:40: DeprecationWarning:
2025-01-06T15:37:54.6059020Z   
2025-01-06T15:37:54.6059160Z   The current CMFGEN model parser is deprecated
2025-01-06T15:37:54.6059310Z 
2025-01-06T15:37:54.6059480Z tardis/model/tests/test_base.py::TestModelFromPaper1Config::test_abundances
2025-01-06T15:37:54.6060000Z   /Users/runner/work/tardis/tardis/tardis/transport/montecarlo/montecarlo_main_loop.py:123: NumbaTypeSafetyWarning:
2025-01-06T15:37:54.6060410Z   
2025-01-06T15:37:54.6060820Z   [1m[1m[1munsafe cast from uint64 to int64. Precision may be lost.[0m[0m[0m
2025-01-06T15:37:54.6061030Z 
2025-01-06T15:37:54.6061110Z tardis/model/tests/test_base.py: 6 warnings
2025-01-06T15:37:54.6061340Z tardis/model/tests/test_csvy_model.py: 2 warnings
2025-01-06T15:37:54.6061630Z tardis/plasma/tests/test_tardis_model_density_config.py: 3 warnings
2025-01-06T15:37:54.6062050Z   /Users/runner/work/tardis/tardis/tardis/model/geometry/radial1d.py:50: DeprecationWarning:
2025-01-06T15:37:54.6062380Z   
2025-01-06T15:37:54.6062530Z   v_inner_boundary < 0, assuming default value
2025-01-06T15:37:54.6062680Z 
2025-01-06T15:37:54.6062750Z tardis/model/tests/test_base.py: 6 warnings
2025-01-06T15:37:54.6062980Z tardis/model/tests/test_csvy_model.py: 2 warnings
2025-01-06T15:37:54.6063280Z tardis/plasma/tests/test_tardis_model_density_config.py: 3 warnings
2025-01-06T15:37:54.6063690Z   /Users/runner/work/tardis/tardis/tardis/model/geometry/radial1d.py:61: DeprecationWarning:
2025-01-06T15:37:54.6064670Z   
2025-01-06T15:37:54.6064900Z   v_outer_boundary < 0, assuming default value
2025-01-06T15:37:54.6065130Z 
2025-01-06T15:37:54.6065580Z tardis/plasma/equilibrium/tests/test_level_populations.py::TestLevelPopulationSolver::test_solve[collisional_rate_solver0-radiative_transitions0]
2025-01-06T15:37:54.6066330Z   /Users/runner/work/tardis/tardis/tardis/plasma/equilibrium/level_populations.py:65: FutureWarning:
2025-01-06T15:37:54.6066860Z   
2025-01-06T15:37:54.6067260Z   A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
2025-01-06T15:37:54.6068080Z   The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.
2025-01-06T15:37:54.6104350Z   
2025-01-06T15:37:54.6105010Z   For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.
2025-01-06T15:37:54.6105730Z 
2025-01-06T15:37:54.6106340Z tardis/plasma/tests/test_complete_plasmas.py: 20 warnings
2025-01-06T15:37:54.6107000Z   /Users/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:37:54.6107520Z   
2025-01-06T15:37:54.6107650Z   Property "atomic_mass" not found
2025-01-06T15:37:54.6108030Z 
2025-01-06T15:37:54.6108610Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-w_epsilon:1e-10-j_blues_norm_factor]
2025-01-06T15:37:54.6109450Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-j_blues_norm_factor]
2025-01-06T15:37:54.6110080Z   /Users/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:37:54.6110870Z   
2025-01-06T15:37:54.6111010Z   Property "j_blues_norm_factor" not found
2025-01-06T15:37:54.6111160Z 
2025-01-06T15:37:54.6111480Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-w_epsilon:1e-10-j_blue_estimator]
2025-01-06T15:37:54.6112270Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-j_blue_estimator]
2025-01-06T15:37:54.6112840Z   /Users/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:37:54.6113170Z   
2025-01-06T15:37:54.6113300Z   Property "j_blue_estimator" not found
2025-01-06T15:37:54.6113440Z 
2025-01-06T15:37:54.6113750Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-w_epsilon:1e-10-volume]
2025-01-06T15:37:54.6114400Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-volume]
2025-01-06T15:37:54.6114960Z   /Users/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:37:54.6115290Z   
2025-01-06T15:37:54.6115410Z   Property "volume" not found
2025-01-06T15:37:54.6115530Z 
2025-01-06T15:37:54.6115840Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-w_epsilon:1e-10-r_inner]
2025-01-06T15:37:54.6116460Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-r_inner]
2025-01-06T15:37:54.6117010Z   /Users/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:37:54.6117340Z   
2025-01-06T15:37:54.6117460Z   Property "r_inner" not found
2025-01-06T15:37:54.6117580Z 
2025-01-06T15:37:54.6117760Z tardis/plasma/tests/test_nlte_solver.py::test_critical_case_dilution_factor_1_root
2025-01-06T15:37:54.6118350Z   /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py:68: NumbaPendingDeprecationWarning:
2025-01-06T15:37:54.6119110Z   
2025-01-06T15:37:54.6120160Z   [1m[1m[1m[1mCode using Numba extension API maybe depending on 'old_style' error-capturing, which is deprecated and will be replaced by 'new_style' in a future release. See details at https://numba.readthedocs.io/en/latest/reference/deprecation.html#deprecation-of-old-style-numba-captured-errors
2025-01-06T15:37:54.6121010Z   Exception origin:
2025-01-06T15:37:54.6121450Z     File "/Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py", line 526, in _sequence_of_arrays
2025-01-06T15:37:54.6121970Z       raise TypeError("%s(): expecting a non-empty tuple of arrays, "
2025-01-06T15:37:54.6122250Z   [0m[0m[0m[0m
2025-01-06T15:37:54.6122340Z 
2025-01-06T15:37:54.6122510Z tardis/plasma/tests/test_plasma_vboundary.py::test_plasma_vboundary[2900-3750]
2025-01-06T15:37:54.6122920Z tardis/plasma/tests/test_plasma_vboundary.py::test_plasma_vboundary[2900-3850]
2025-01-06T15:37:54.6123310Z tardis/plasma/tests/test_plasma_vboundary.py::test_plasma_vboundary[2900-3900]
2025-01-06T15:37:54.6123740Z   /Users/runner/work/tardis/tardis/tardis/model/geometry/radial1d.py:71: UserWarning:
2025-01-06T15:37:54.6124030Z   
2025-01-06T15:37:54.6124240Z   Requesting inner boundary below inner shell. Extrapolating the inner cell
2025-01-06T15:37:54.6124460Z 
2025-01-06T15:37:54.6124630Z tardis/spectrum/tests/test_spectrum_solver.py::TestSpectrumSolver::test_solve
2025-01-06T15:37:54.6125020Z   /Users/runner/work/tardis/tardis/tardis/spectrum/base.py:70: UserWarning:
2025-01-06T15:37:54.6125290Z   
2025-01-06T15:37:54.6125620Z   SpectrumSolver.spectrum_virtual_packets is zero. Please run the montecarlo simulation with no_of_virtual_packets > 0
2025-01-06T15:37:54.6125940Z 
2025-01-06T15:37:54.6126110Z tardis/spectrum/tests/test_spectrum_solver.py::TestSpectrumSolver::test_solve
2025-01-06T15:37:54.6126600Z   /Users/runner/work/tardis/tardis/tardis/spectrum/base.py:111: UserWarning:
2025-01-06T15:37:54.6126870Z   
2025-01-06T15:37:54.6127220Z   SpectrumSolver.integrator: The FormalIntegrator is not yet available.Please run the montecarlo simulation at least once.
2025-01-06T15:37:54.6127550Z 
2025-01-06T15:37:54.6127720Z tardis/transport/montecarlo/tests/test_vpacket.py::test_trace_vpacket_volley
2025-01-06T15:37:54.6128040Z   <string>:3: NumbaPendingDeprecationWarning:
2025-01-06T15:37:54.6128240Z   
2025-01-06T15:37:54.6129070Z   [1m[1mCode using Numba extension API maybe depending on 'old_style' error-capturing, which is deprecated and will be replaced by 'new_style' in a future release. See details at https://numba.readthedocs.io/en/latest/reference/deprecation.html#deprecation-of-old-style-numba-captured-errors
2025-01-06T15:37:54.6129850Z   Exception origin:
2025-01-06T15:37:54.6130150Z     File "/Users/runner/micromamba/envs/tardis/lib/python3.12/inspect.py", line 3180, in _bind
2025-01-06T15:37:54.6130480Z       raise TypeError(msg) from None
2025-01-06T15:37:54.6130680Z   [0m[0m
2025-01-06T15:37:54.6130760Z 
2025-01-06T15:37:54.6131020Z tardis/visualization/tools/tests/test_sdec_plot.py::TestSDECPlotter::test_mpl_image[plotter_generate_plot_mpl4]
2025-01-06T15:37:54.6131570Z   /Users/runner/work/tardis/tardis/tardis/visualization/tools/sdec_plot.py:1225: RuntimeWarning:
2025-01-06T15:37:54.6131910Z   
2025-01-06T15:37:54.6132630Z   More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`). Consider using `matplotlib.pyplot.close()`.
2025-01-06T15:37:54.6133340Z 
2025-01-06T15:37:54.6133620Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6134240Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6134860Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6135970Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6136590Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6137210Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6137840Z   /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/traitlets/traitlets.py:1385: DeprecationWarning:
2025-01-06T15:37:54.6138230Z   
2025-01-06T15:37:54.6138450Z   Passing unrecognized arguments to super(Checkbox).__init__(icon='lock').
2025-01-06T15:37:54.6138810Z   object.__init__() takes exactly one argument (the instance to initialize)
2025-01-06T15:37:54.6139210Z   This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
2025-01-06T15:37:54.6139470Z 
2025-01-06T15:37:54.6139750Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6140360Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6140970Z   /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/traitlets/traitlets.py:1385: DeprecationWarning:
2025-01-06T15:37:54.6141360Z   
2025-01-06T15:37:54.6141560Z   Passing unrecognized arguments to super(FloatText).__init__(min=0).
2025-01-06T15:37:54.6141900Z   object.__init__() takes exactly one argument (the instance to initialize)
2025-01-06T15:37:54.6142300Z   This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
2025-01-06T15:37:54.6142560Z 
2025-01-06T15:37:54.6142970Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:37:54.6143580Z   /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/traitlets/traitlets.py:1385: DeprecationWarning:
2025-01-06T15:37:54.6143970Z   
2025-01-06T15:37:54.6144170Z   Passing unrecognized arguments to super(HTML).__init__(indent=True).
2025-01-06T15:37:54.6144520Z   object.__init__() takes exactly one argument (the instance to initialize)
2025-01-06T15:37:54.6144910Z   This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
2025-01-06T15:37:54.6145170Z 
2025-01-06T15:37:54.6145510Z tardis/visualization/widgets/tests/test_line_info.py::TestLineInfoWidgetData::test_get_species_interactions[wavelength_range0-packet_out_nu]
2025-01-06T15:37:54.6146160Z   /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/qgridnext/grid.py:820: DeprecationWarning:
2025-01-06T15:37:54.6146540Z   
2025-01-06T15:37:54.6146810Z   QgridWidget._precision_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead.
2025-01-06T15:37:54.6147110Z 
2025-01-06T15:37:54.6147310Z tardis/visualization/widgets/tests/test_shell_info.py::TestHDFShellInfo::test_shells_data
2025-01-06T15:37:54.6147750Z   /Users/runner/work/tardis/tardis/tardis/io/util.py:289: PerformanceWarning:
2025-01-06T15:37:54.6148030Z   
2025-01-06T15:37:54.6148140Z   
2025-01-06T15:37:54.6148360Z   your performance may suffer as PyTables will pickle object types that it cannot
2025-01-06T15:37:54.6148800Z   map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index([0], dtype='int64')]
2025-01-06T15:37:54.6149080Z 
2025-01-06T15:37:54.6149290Z tardis/visualization/widgets/tests/test_shell_info.py::TestHDFShellInfo::test_shells_data
2025-01-06T15:37:54.6149730Z   /Users/runner/work/tardis/tardis/tardis/io/util.py:292: PerformanceWarning:
2025-01-06T15:37:54.6150010Z   
2025-01-06T15:37:54.6150110Z   
2025-01-06T15:37:54.6150340Z   your performance may suffer as PyTables will pickle object types that it cannot
2025-01-06T15:37:54.6150720Z   map directly to c-types [inferred_type->mixed,key->values] [items->None]
2025-01-06T15:37:54.6151040Z 
2025-01-06T15:37:54.6151270Z tardis/visualization/widgets/tests/test_shell_info.py::TestHDFShellInfo::test_shells_data
2025-01-06T15:37:54.6151850Z   /Users/runner/work/tardis/tardis/tardis/spectrum/formal_integral.py:398: UserWarning:
2025-01-06T15:37:54.6152180Z   
2025-01-06T15:37:54.6152390Z   The number of interpolate_shells was not specified. The value was set to 80.
2025-01-06T15:37:54.6152610Z 
2025-01-06T15:37:54.6152780Z -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
2025-01-06T15:37:54.6153010Z 
2025-01-06T15:37:54.6153140Z ---------- coverage: platform darwin, python 3.12.4-final-0 ----------
2025-01-06T15:37:54.6153430Z Coverage HTML written to dir coverage_html_report
2025-01-06T15:37:54.6153670Z Coverage XML written to file coverage.xml
2025-01-06T15:37:54.6153830Z 
2025-01-06T15:37:54.6153930Z =========================== short test summary info ============================
2025-01-06T15:37:54.6154530Z ERROR tardis/plasma/tests/test_nlte_solver.py::test_critical_case_dilution_factor_1_root - numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6155310Z [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6155770Z [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:37:54.6156090Z  
2025-01-06T15:37:54.6156280Z  >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:37:54.6156520Z  
2025-01-06T15:37:54.6156650Z There are 2 candidate implementations:
2025-01-06T15:37:54.6156890Z [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.6157200Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.6157640Z     With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.6158200Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6158540Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6159040Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.6159350Z     
2025-01-06T15:37:54.6159590Z     >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.6159870Z     
2025-01-06T15:37:54.6160010Z    There are 2 candidate implementations:
2025-01-06T15:37:54.6160260Z    [1m      - Of which 2 did not match due to:
2025-01-06T15:37:54.6160580Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6161020Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.6161410Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6161940Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6162550Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6162950Z    [0m
2025-01-06T15:37:54.6163220Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.6163790Z    [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.6164210Z    [0m
2025-01-06T15:37:54.6164330Z    [1m
2025-01-06T15:37:54.6164680Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.6165080Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.6165340Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.6165590Z    [0m            [1m^[0m[0m
2025-01-06T15:37:54.6165770Z [0m
2025-01-06T15:37:54.6166100Z   raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.6166640Z [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.6166950Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.6167370Z     With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.6167720Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6168060Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6168520Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.6168830Z     
2025-01-06T15:37:54.6169050Z     >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.6169330Z     
2025-01-06T15:37:54.6169470Z    There are 2 candidate implementations:
2025-01-06T15:37:54.6169710Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.6170030Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6170460Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.6170840Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6171360Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6171960Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6172400Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.6172710Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6173230Z            With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:37:54.6173620Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6174130Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6174730Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6175110Z    [0m
2025-01-06T15:37:54.6175500Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.6176060Z    [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.6176470Z    [0m
2025-01-06T15:37:54.6176600Z    [1m
2025-01-06T15:37:54.6176940Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.6177320Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.6177580Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.6177820Z    [0m            [1m^[0m[0m
2025-01-06T15:37:54.6177990Z [0m
2025-01-06T15:37:54.6178310Z   raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.6178690Z [0m
2025-01-06T15:37:54.6178970Z [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:37:54.6179580Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:37:54.6180010Z [0m
2025-01-06T15:37:54.6180140Z [1m
2025-01-06T15:37:54.6180430Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:37:54.6180790Z [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:37:54.6180990Z     <source elided>
2025-01-06T15:37:54.6181090Z 
2025-01-06T15:37:54.6181250Z [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:37:54.6181640Z [0m    [1m^[0m[0m
2025-01-06T15:37:54.6181740Z 
2025-01-06T15:37:54.6182050Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.6182730Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.6183160Z [0m
2025-01-06T15:37:54.6183510Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.6184240Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.6184670Z [0m
2025-01-06T15:37:54.6184790Z [1m
2025-01-06T15:37:54.6185080Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:37:54.6185510Z [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:37:54.6185780Z     <source elided>
2025-01-06T15:37:54.6185870Z 
2025-01-06T15:37:54.6186040Z [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:37:54.6186320Z [0m            [1m^[0m[0m
2025-01-06T15:37:54.6186850Z ERROR tardis/plasma/tests/test_nlte_solver.py::test_critical_case_dilution_factor_1_lu - numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6187480Z [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6187930Z [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:37:54.6188250Z  
2025-01-06T15:37:54.6188430Z  >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:37:54.6188660Z  
2025-01-06T15:37:54.6188860Z There are 2 candidate implementations:
2025-01-06T15:37:54.6189090Z [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.6189400Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.6189810Z     With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.6190160Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6190490Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6190950Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.6191250Z     
2025-01-06T15:37:54.6191470Z     >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.6191740Z     
2025-01-06T15:37:54.6191880Z    There are 2 candidate implementations:
2025-01-06T15:37:54.6192120Z    [1m      - Of which 2 did not match due to:
2025-01-06T15:37:54.6192420Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6192850Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.6193220Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6193720Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6194310Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6194690Z    [0m
2025-01-06T15:37:54.6194950Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.6195510Z    [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.6195910Z    [0m
2025-01-06T15:37:54.6196030Z    [1m
2025-01-06T15:37:54.6196370Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.6197120Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.6197370Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.6197600Z    [0m            [1m^[0m[0m
2025-01-06T15:37:54.6197770Z [0m
2025-01-06T15:37:54.6198090Z   raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.6198490Z [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.6198790Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.6199200Z     With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.6199550Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6199870Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6200330Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.6200650Z     
2025-01-06T15:37:54.6200860Z     >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.6201130Z     
2025-01-06T15:37:54.6201260Z    There are 2 candidate implementations:
2025-01-06T15:37:54.6201500Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.6201800Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6202230Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.6202590Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6203100Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6203690Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6204270Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.6209930Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6210550Z            With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:37:54.6210950Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6211450Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6212040Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6212420Z    [0m
2025-01-06T15:37:54.6212690Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.6213250Z    [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.6213660Z    [0m
2025-01-06T15:37:54.6213790Z    [1m
2025-01-06T15:37:54.6214120Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.6214500Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.6214750Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.6214980Z    [0m            [1m^[0m[0m
2025-01-06T15:37:54.6215150Z [0m
2025-01-06T15:37:54.6215590Z   raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.6215960Z [0m
2025-01-06T15:37:54.6216230Z [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:37:54.6216830Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:37:54.6217260Z [0m
2025-01-06T15:37:54.6217500Z [1m
2025-01-06T15:37:54.6217780Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:37:54.6218350Z [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:37:54.6218540Z     <source elided>
2025-01-06T15:37:54.6218630Z 
2025-01-06T15:37:54.6218780Z [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:37:54.6219030Z [0m    [1m^[0m[0m
2025-01-06T15:37:54.6219130Z 
2025-01-06T15:37:54.6219420Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.6220080Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.6220500Z [0m
2025-01-06T15:37:54.6220850Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.6221500Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.6221930Z [0m
2025-01-06T15:37:54.6222050Z [1m
2025-01-06T15:37:54.6222330Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:37:54.6222750Z [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:37:54.6223000Z     <source elided>
2025-01-06T15:37:54.6223090Z 
2025-01-06T15:37:54.6223250Z [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:37:54.6223510Z [0m            [1m^[0m[0m
2025-01-06T15:37:54.6224050Z ERROR tardis/plasma/tests/test_nlte_solver.py::test_critical_case_dilution_factor_0_root - numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6224690Z [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6225130Z [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:37:54.6225530Z  
2025-01-06T15:37:54.6225710Z  >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:37:54.6225950Z  
2025-01-06T15:37:54.6226080Z There are 2 candidate implementations:
2025-01-06T15:37:54.6226310Z [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.6226610Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.6227030Z     With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.6227380Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6227710Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6228160Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.6228470Z     
2025-01-06T15:37:54.6228700Z     >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.6228980Z     
2025-01-06T15:37:54.6229120Z    There are 2 candidate implementations:
2025-01-06T15:37:54.6229360Z    [1m      - Of which 2 did not match due to:
2025-01-06T15:37:54.6229670Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6230090Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.6230480Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6231020Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6231600Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6231980Z    [0m
2025-01-06T15:37:54.6232250Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.6232810Z    [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.6233310Z    [0m
2025-01-06T15:37:54.6233430Z    [1m
2025-01-06T15:37:54.6233760Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.6234140Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.6234390Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.6234620Z    [0m            [1m^[0m[0m
2025-01-06T15:37:54.6234790Z [0m
2025-01-06T15:37:54.6235100Z   raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.6235500Z [1m  - Of which 1 did not match due to:
2025-01-06T15:37:54.6235800Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:37:54.6236210Z     With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:37:54.6236550Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6236880Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:37:54.6237330Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:37:54.6237640Z     
2025-01-06T15:37:54.6237860Z     >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:37:54.6238130Z     
2025-01-06T15:37:54.6238260Z    There are 2 candidate implementations:
2025-01-06T15:37:54.6238500Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.6238810Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6239230Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:37:54.6239590Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6240160Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6240750Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6241170Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:37:54.6241480Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:37:54.6241920Z            With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:37:54.6242300Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:37:54.6242800Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:37:54.6243380Z      raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:37:54.6243770Z    [0m
2025-01-06T15:37:54.6244030Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:37:54.6244590Z    [0m[1mDuring: typing of call at /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:37:54.6244990Z    [0m
2025-01-06T15:37:54.6245110Z    [1m
2025-01-06T15:37:54.6245440Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:37:54.6245810Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:37:54.6246060Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:37:54.6246290Z    [0m            [1m^[0m[0m
2025-01-06T15:37:54.6246460Z [0m
2025-01-06T15:37:54.6246780Z   raised from /Users/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:37:54.6247160Z [0m
2025-01-06T15:37:54.6247420Z [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:37:54.6248020Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:37:54.6248520Z [0m
2025-01-06T15:37:54.6248640Z [1m
2025-01-06T15:37:54.6248920Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:37:54.6249270Z [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:37:54.6249460Z     <source elided>
2025-01-06T15:37:54.6249550Z 
2025-01-06T15:37:54.6249710Z [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:37:54.6249960Z [0m    [1m^[0m[0m
2025-01-06T15:37:54.6250060Z 
2025-01-06T15:37:54.6250360Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.6251040Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.6251470Z [0m
2025-01-06T15:37:54.6251830Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x127bd3ec0>))[0m
2025-01-06T15:37:54.6252500Z [0m[1mDuring: typing of call at /Users/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:37:54.6252920Z [0m
2025-01-06T15:37:54.6253040Z [1m
2025-01-06T15:37:54.6253330Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:37:54.6253760Z [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:37:54.6254020Z     <source elided>
2025-01-06T15:37:54.6254120Z 
2025-01-06T15:37:54.6254280Z [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:37:54.6254560Z [0m            [1m^[0m[0m
2025-01-06T15:37:54.6254880Z = 1435 passed, 329 skipped, 2 deselected, 20 xfailed, 7 xpassed, 92 warnings, 3 errors in 670.90s (0:11:10) =
2025-01-06T15:37:57.3995930Z ##[error]Process completed with exit code 1.
